python/cudf/cudf/comm/gpuarrow.py

```diff
@@ -141,15 +142,14 @@ def _make_dictionary_series(self):
         return self._series.copy(deep=False)
 
 
-def gpu_view_as(buf, dtype, shape=None, strides=None):
+def gpu_view_as(nbytes, buf, dtype, shape=None, strides=None):
     ptr = numba.cuda.cudadrv.driver.device_pointer(buf.to_numba())
-    return rmm.device_array_from_ptr(
-        ptr, buf.size // dtype.itemsize, dtype=dtype
-    )
+    arr = rmm.device_array_from_ptr(ptr, nbytes // dtype.itemsize, dtype=dtype)
+    arr.gpu_data._obj = buf
```

Comment:
Is this necessary to prevent a use after free bug?

Reply:
Yup